pkg/kubelet/container/container_gc.go

```diff
@@ -39,7 +39,7 @@ type ContainerGCPolicy struct {
 // Implementation is thread-compatible.
 type ContainerGC interface {
 	// Garbage collect containers.
-	GarbageCollect() error
+	GarbageCollect(allSourcesReady bool) error
```

Comment:
We don't want to refactor the code extensively due to reason that @Random-Liu said. Making this part of the ContainerGCPolicy is even uncleaner, since the state changes dynamically (at least during kubelet startup).